src/Simplex/Messaging/Client.hs

```diff
@@ -78,35 +80,52 @@ data Request = Request
     responseVar :: TMVar (Either SMPClientError Cmd)
   }
 
-getSMPClient :: SMPServer -> SMPClientConfig -> TBQueue SMPServerTransmission -> IO SMPClient
+getSMPClient :: SMPServer -> SMPClientConfig -> TBQueue SMPServerTransmission -> IO () -> IO SMPClient
```

Comment:
I understand why `IO ()` is added, but it looks a bit hacky. Make a type synonym e.g. `type DisconnectHandler = IO ()`?

Reply:
On the other hand it's clear from arguments

Reply:
it's quite common to pass continuation like this - all bracket functions would accept `IO a` as a parameter, no need to create a synonym for it - `IO a` always means "something to do with side effects and that returns a as a result". There is no real difference between IO () as return type and IO () as parameter - both define an action that can be performed.